File: test_integration.py
#!/usr/bin/env python3
"""
Simple integration test for frontend-backend communication
"""
import requests
import json
import sys

BACKEND_URL = 'http://localhost:8000/api'
FRONTEND_URL = 'http://localhost:3000'

def test_backend_health():
    """Test backend health endpoint"""
    try:
        print('Testing backend health endpoint...')
        response = requests.get(f'{BACKEND_URL}/health', timeout=5)
        response.raise_for_status()
        data = response.json()
        print(f'✅ Backend health check passed: {data}')
        return True
    except Exception as e:
        print(f'❌ Backend health check failed: {e}')
        return False

def test_backend_analyze():
    """Test backend analyze endpoint"""
    try:
        print('Testing backend analyze endpoint...')
        
        test_request = {
            "products": [
                {
                    "name": "Corn Seeds",
                    "quantity": 100,
                    "max_price": 150.0,
                    "urgency": "medium"
                }
            ],
            "farm_location": {
                "state": "Iowa",
                "county": "Polk",
                "zip_code": "50309"
            }
        }
        
        response = requests.post(f'{BACKEND_URL}/analyze', json=test_request, timeout=30)
        response.raise_for_status()
        data = response.json()
        
        print('✅ Backend analyze endpoint passed')
        print(f'Response structure:')
        print(f'  - Product analyses: {len(data.get("product_analyses", []))}')
        print(f'  - Overall budget: {"✓" if data.get("overall_budget") else "✗"}')
        print(f'  - Data quality report: {"✓" if data.get("data_quality_report") else "✗"}')
        return True
    except Exception as e:
        print(f'❌ Backend analyze endpoint failed: {e}')
        if hasattr(e, 'response') and e.response is not None:
            try:
                error_data = e.response.json()
                print(f'Error details: {error_data}')
            except:
                print(f'Error response: {e.response.text}')
        return False

def test_frontend_access():
    """Test frontend accessibility"""
    try:
        print('Testing frontend accessibility...')
        response = requests.get(FRONTEND_URL, timeout=5)
        response.raise_for_status()
        print('✅ Frontend is accessible')
        return True
    except Exception as e:
        print(f'❌ Frontend access failed: {e}')
        return False

def main():
    """Run integration tests"""
    print('🚀 Starting integration tests...\n')
    
    backend_health = test_backend_health()
    print()
    
    backend_analyze = test_backend_analyze()
    print()
    
    frontend_access = test_frontend_access()
    print()
    
    print('📊 Test Results:')
    print(f'Backend Health: {"✅" if backend_health else "❌"}')
    print(f'Backend Analyze: {"✅" if backend_analyze else "❌"}')
    print(f'Frontend Access: {"✅" if frontend_access else "❌"}')
    
    all_passed = backend_health and backend_analyze and frontend_access
    print(f'\n{"🎉" if all_passed else "⚠️"} Integration test {"PASSED" if all_passed else "FAILED"}')
    
    if all_passed:
        print('\n✨ Your frontend and backend are successfully integrated!')
        print('🌐 Frontend: http://localhost:3000')
        print('🔧 Backend API: http://localhost:8000/api')
        print('📚 API Docs: http://localhost:8000/docs')
    else:
        print('\n🔧 Troubleshooting tips:')
        if not backend_health:
            print('  - Check if backend server is running on port 8000')
        if not backend_analyze:
            print('  - Check backend logs for analysis errors')
        if not frontend_access:
            print('  - Check if frontend server is running on port 3000')
    
    return 0 if all_passed else 1

if __name__ == '__main__':
    sys.exit(main())

File: test_integration.js
// Simple integration test for frontend-backend communication
const axios = require('axios');

const BACKEND_URL = 'http://localhost:8000/api';
const FRONTEND_URL = 'http://localhost:3000';

async function testBackendHealth() {
  try {
    console.log('Testing backend health endpoint...');
    const response = await axios.get(`${BACKEND_URL}/health`);
    console.log('✅ Backend health check passed:', response.data);
    return true;
  } catch (error) {
    console.log('❌ Backend health check failed:', error.message);
    return false;
  }
}

async function testBackendAnalyze() {
  try {
    console.log('Testing backend analyze endpoint...');
    
    const testRequest = {
      products: [
        {
          name: "Corn Seeds",
          quantity: 100,
          max_price: 150.0,
          urgency: "medium"
        }
      ],
      farm_location: {
        state: "Iowa",
        county: "Polk",
        zip_code: "50309"
      }
    };
    
    const response = await axios.post(`${BACKEND_URL}/analyze`, testRequest);
    console.log('✅ Backend analyze endpoint passed');
    console.log('Response structure:', {
      product_analyses: response.data.product_analyses?.length || 0,
      overall_budget: !!response.data.overall_budget,
      data_quality_report: !!response.data.data_quality_report
    });
    return true;
  } catch (error) {
    console.log('❌ Backend analyze endpoint failed:', error.response?.data || error.message);
    return false;
  }
}

async function testFrontendAccess() {
  try {
    console.log('Testing frontend accessibility...');
    const response = await axios.get(FRONTEND_URL, { timeout: 5000 });
    console.log('✅ Frontend is accessible');
    return true;
  } catch (error) {
    console.log('❌ Frontend access failed:', error.message);
    return false;
  }
}

async function runIntegrationTests() {
  console.log('🚀 Starting integration tests...\n');
  
  const backendHealth = await testBackendHealth();
  console.log('');
  
  const backendAnalyze = await testBackendAnalyze();
  console.log('');
  
  const frontendAccess = await testFrontendAccess();
  console.log('');
  
  console.log('📊 Test Results:');
  console.log(`Backend Health: ${backendHealth ? '✅' : '❌'}`);
  console.log(`Backend Analyze: ${backendAnalyze ? '✅' : '❌'}`);
  console.log(`Frontend Access: ${frontendAccess ? '✅' : '❌'}`);
  
  const allPassed = backendHealth && backendAnalyze && frontendAccess;
  console.log(`\n${allPassed ? '🎉' : '⚠️'} Integration test ${allPassed ? 'PASSED' : 'FAILED'}`);
  
  if (allPassed) {
    console.log('\n✨ Your frontend and backend are successfully integrated!');
    console.log('🌐 Frontend: http://localhost:3000');
    console.log('🔧 Backend API: http://localhost:8000/api');
    console.log('📚 API Docs: http://localhost:8000/docs');
  }
}

// Run the tests
runIntegrationTests().catch(console.error);